src/main/java/software/amazon/awssdk/crt/mqtt/MqttConnection.java

```diff
@@ -144,29 +119,78 @@ public void onSuccess() {
             }
 
             @Override
-            public void onFailure(String reason) {
-                Throwable cause = new MqttException(reason);
-                future.completeExceptionally(cause);
+            public void onFailure(Throwable reason) {
+                future.completeExceptionally(reason);
```

Comment:
I DID REAL GOOD

this method name is funny, that's my joke

Reply:
Yeah, welcome to Java!